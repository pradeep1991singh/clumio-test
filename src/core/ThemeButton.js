import React, { useContext } from "react";
import { ThemeContext } from "./ThemeContext";

const ThemeButton = () => {
  const { theme, setTheme } = useContext(ThemeContext);

  return (
    <div className="transition duration-500 ease-in-out py-2">
      <button
        onClick={() => setTheme(theme === "dark" ? "light" : "dark")}
        className="dark:text-gray-500 cursor-pointer rounded px-3 py-1 border"
      >
        Switch to {theme === "dark" ? "light" : "dark"}
      </button>
    </div>
  );
};

export default ThemeButton;
