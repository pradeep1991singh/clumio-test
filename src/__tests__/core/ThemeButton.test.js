import React from "react";
import { render } from "@testing-library/react";
import ThemeButton from "../../core/ThemeButton";

describe("<ThemeButton />", () => {
  test("Snapshot", () => {
    const { asFragment } = render(<ThemeButton />);
    expect(asFragment()).toMatchSnapshot();
  });
});
