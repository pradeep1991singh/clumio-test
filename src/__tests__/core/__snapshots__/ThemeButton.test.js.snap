// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ThemeButton /> Snapshot 1`] = `
<DocumentFragment>
  <div
    class="transition duration-500 ease-in-out py-2"
  >
    <button
      class="dark:text-gray-500 cursor-pointer rounded px-3 py-1 border"
    >
      Switch to dark
    </button>
  </div>
</DocumentFragment>
`;
